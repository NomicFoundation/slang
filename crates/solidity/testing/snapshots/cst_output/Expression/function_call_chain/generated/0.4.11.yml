# This file is generated automatically by infrastructure scripts. Please don't edit by hand.

Source: >
  1  │ x()()                                                                            │ 0..5

Errors: []

Tree:
  - Expression (Rule): # 0..5 "x()()"
      - FunctionCallExpression (Rule): # 0..5 "x()()"
          - FunctionCallExpression (Rule): # 0..3 "x()"
              - PrimaryExpression (Rule): # 0..1 "x"
                  - Identifier (Token): "x" # 0..1
              - ArgumentList (Rule): # 1..3 "()"
                  - _DELIMITEDBY (Rule): # 1..3 "()"
                      - OpenParen (Token): "(" # 1..2
                      - CloseParen (Token): ")" # 2..3
          - ArgumentList (Rule): # 3..5 "()"
              - _DELIMITEDBY (Rule): # 3..5 "()"
                  - OpenParen (Token): "(" # 3..4
                  - CloseParen (Token): ")" # 4..5
