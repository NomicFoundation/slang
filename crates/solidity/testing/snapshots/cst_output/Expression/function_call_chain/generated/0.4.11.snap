# This file is generated automatically by infrastructure scripts. Please don't edit by hand.

Errors: 0

█ Rule(Expression) - (0..5)                       │
└──█ Rule(FunctionCallExpression) - (0..5)        │
   ├──█ Rule(FunctionCallExpression) - (0..3)     │
   │  ├──█ Rule(PrimaryExpression) - (0..1)       │
   │  │  └──► Token(Identifier) - (0..1)          │ x
   │  └──█ Group - (1..3)                         │
   │     └──█ Rule(ArgumentList) - (1..3)         │
   │        ├──► Token(OpenParen) - (1..2)        │ (
   │        └──► Token(CloseParen) - (2..3)       │ )
   └──█ Group - (3..5)                            │
      └──█ Rule(ArgumentList) - (3..5)            │
         ├──► Token(OpenParen) - (3..4)           │ (
         └──► Token(CloseParen) - (4..5)          │ )
